streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedStreamsTasks.java

```diff
@@ -511,26 +529,13 @@ public void shutdown(final boolean clean) {
         super.shutdown(clean);
     }
 
-    @Override
-    public boolean isEmpty() throws IllegalStateException {
-        if (restoring.isEmpty() && !restoringByPartition.isEmpty()) {
-            log.error("Assigned stream tasks in an inconsistent state: the set of restoring tasks is empty but the " +
```

Comment:
Same here

Reply:
ditto above, this was part of the strict safety enforcement that we want to drop in 2.4 in favor of just clearing everything